docarray/array/mixins/io/pushpull.py

```diff
@@ -52,24 +71,35 @@ def push(self, token: str, show_progress: bool = False) -> None:
               persistence. Only use this full temporary transmission/storage/clipboard.
 
         :param token: a key that later can be used for retrieve this :class:`DocumentArray`.
+        :param id: an unique identifier that returned from previous :meth:`.push`
```

Comment:
> returned from previous :meth:`.push`
what does this mean?

Reply:
still confused, what is `id`? This variable is too general.

is it about filename? or a credential? to help me better understand it, could you copy paste one of your "id" so that I can see it?

Reply:
`id` is the id of the database entry on the Hubble.

Here is a demo:
```python
da = DocumentArray([Document(text='hello')])
res = da.push() // authenticated users can push without token
```
`res` is dict which is response from Hubble API
`{'_id': '62432cd4665a8ed1a5f87d6a', 'visibility': 'private', ...}`

`62432cd4665a8ed1a5f87d6a` is the `id` I am talking about.
Later on, I can use this `id` to push or pull old DA.

```python
da = DocumentArray([Document(text='hello world')])
da.push(id='62432cd4665a8ed1a5f87d6a') // replacing old DA
// or
DocumentArray.pull(id='62432cd4665a8ed1a5f87d6a')
```

Yes, it is confusing because now we have 2 ways to do `push/pull`.
It is needed because logged-in users don't have to come up with tokens to protect their DA. They can just save `id` of it somewhere. No one can see or edit their DA by default.
Also, 3rd party programs (e.g, Finetuner) can manage user DAs if they have user authentication.

Sorry for the confusion, I hope it helps. Maybe we need to rename the parameter and/or improve documentation.